src/design/DesignBuilder.ts

```diff
@@ -15,14 +15,15 @@ type SharedEntitiyProperties = {
     name?: string;
     extendedData?: {
         customStrings: Array<string>;
-    };
+    } & ExtendedData;
```

Comment:
This doesnt seem to make sense as wouldnt you end up with a type that looks like:

```
extendedData: {
  customStrings: Array<string>;
  name: string;
  value: string;
}
```
?

Maybe the best solution would be to make all values of `extendedData` an `Array<string>`

```
extendedData: {
   [name: string]: Array<string>;
}
```

Then ensure all values are wrapped in an array.